### DISCUSSION TITLE
[Animation] repetitions not working after second time calling play()

### DISCUSSION THREAD

**Discussion by futahei** [google/model-viewer#3618] [Q&A]
### Description

`repetitions`(`play()` argument) is not working on the second or subsequent calls.

#### Live Demo
https://dull-discreet-antelope.glitch.me
https://glitch.com/edit/#!/dull-discreet-antelope

### Version
- recent commit ([8e52e80528f37d267e1717a89a2da89c71e52bf3](https://github.com/google/model-viewer/commit/8e52e80528f37d267e1717a89a2da89c71e52bf3))

### Browser Affected

- [x] Chrome, version: 101.0.4951.67
- [x] Edge
- [ ] Firefox
- [ ] IE
- [x] Safari

### OS

- [ ] Android
- [x] iOS
- [ ] Linux
- [ ] MacOS
- [x] Windows

**Comment by elalish** [comment]
Is this a regression?

**Comment by futahei** [comment]
I'm sorry. The shared link was incorrect.
Please see the code.
https://glitch.com/edit/#!/dull-discreet-antelope

**Comment by alexdaube** [comment]
Couple of other behaviours I noticed along side the one above.

#### **`pingpong` with repetitions seems to have no effect**:
```play({ repetitions: 1, pinpong: true })```
https://glitch.com/edit/#!/grave-bouncy-family

Unless I don't fully understand what the pingpong functionality is supposed to do, I would expect the Robot to sit down and then get back up here.

#### Changing the animation programatically before calling play always set repetitions to `Infinity`:
```
  mv.animationName = 'Dance'
  mv.play({ repetitions: 1 });
```
https://glitch.com/edit/#!/bramble-power-silkworm

For this code to work, `mv.play` needs to be called inside a `setTimeout`.

**Comment by futahei** [comment]
@alexdaube
I think it was a specification on model-viewer.
> Changing the animation programatically before calling play always set repetitions to Infinity:
> ```
> mv.animationName = 'Dance'
> mv.play({ repetitions: 1 });
> ```
You can fix it by calling `mv.updateComplete`. See below,
```
mv.animationName = 'Dance'
await mv.update Complete
mv.play({ repetitions: 1 });
```
